api/src/libs/messaging.ts

```diff
@@ -46,10 +45,55 @@ export const contactsForProfile = async (profileId: number): Promise<string[]> =
   }
 }
 
+export const profileDetails = async (profileId: number): Promise<string> => {
+
+  try {
+    const dm = new DataManager(shared.pgPool);
+    const { ProfileModel } = dm;
+    const profiles = await ProfileModel.findById(profileId);
+    const profileName = profiles.name;
+    return profileName;
+  } catch (err) {
+    const message = `Unable to fetch profile ${profileId}`;
+    logger.error(`${message}, err = ${err.message}`);
+
+    return '';
+  }
+}
+
+export const updateEmailContent = async (buff: string, to: string[], profileName: string, contactNames: string[]): Promise<string> => {
+  try {
+    let emailContent: string;
+
+    const mapObj = {
+      POName: contactNames[0],
+      TCName: contactNames[1],
+      POEmail: to[0],
+      TCEmail: to[1],
+      projectName: profileName,
+    };
+
+    const re = new RegExp(Object.keys(mapObj).join('|'),'gi');
+    emailContent = buff.replace(re, matched => {
```

Comment:
I wonder if this would work:

```
emailContent = buff.replace((re, matched) => mapObj[matched]);
```

Reply:
This would need a minor adjustment as .replace() flags as only one argument if `(re, matched)` is wrapped in (). However, removing that appears to return the keys vs values. Not entirely sure why to be honest though.

Reply:
I think it was an old bug. Rebuilt the api and have had no further issues.